frontend/src/pages/Common/Budget.js

```diff
@@ -51,232 +93,189 @@ export default class Budget extends React.Component {
         })
       )
     ];
-
-    return distinctOrganizations.map((organization, index) => {
+    return distinctOrganizations.map(organization => {
       return (
         <MenuItem key={`budget-${organization}`} value={organization}>
           {organization}
         </MenuItem>
       );
     });
-  }
+  }, []);
 
-  deleteBudgetFromList(projectedBudgets, deletedProjectedBudgets, budgetToDelete) {
-    this.props.storeDeletedProjectedBudget(this.addBudget(deletedProjectedBudgets, budgetToDelete));
-    this.setCurrency(this.state.currency);
-    this.setOrganization(this.state.organization);
-  }
-
-  updateSavable(organization, currency) {
-    const isSaveable = !this.props.projectedBudgets.some(
-      x => x.organization === organization && x.currencyCode === currency
-    );
-
-    if (
-      this.state.isSaveable !== isSaveable ||
-      this.state.organization !== organization ||
-      this.state.currency !== currency
-    ) {
-      this.setState({
-        organization: organization,
-        currency: currency,
-        isSaveable
-      });
-    }
-  }
-  setOrganization(organization) {
-    this.updateSavable(organization, this.state.currency);
-  }
-
-  setCurrency(currency) {
-    this.updateSavable(this.state.organization, currency);
-  }
-
-  editProjectedBudget(budgets, budget, budgetAmountEdit) {
-    const updateIndex = budgets.findIndex(
-      x => x.organization === budget.organization && x.currencyCode === budget.currencyCode
-    );
-
-    budgets[updateIndex].value = budgetAmountEdit;
-    this.props.storeProjectedBudget(budgets);
-    this.setState({
-      edit: false,
-      editIndex: -1
+  const getCurrencyMenuItems = useCallback(currencies => {
+    return currencies.map(currency => {
+      return (
+        <MenuItem key={currency.primaryText} value={currency.value}>
+          {currency.primaryText}
+        </MenuItem>
+      );
     });
-  }
+  }, []);
 
-  addBudget(budgets, budgetToAdd) {
-    budgets.push(budgetToAdd);
-    return budgets;
-  }
+  const saveProjectedBudget = useCallback(() => {
+    const projectedBudgetsCopy = projectedBudgets;
+    projectedBudgetsCopy.push({
```

Comment:
this is a sideEffect and has to be done via immutable.js in a reducer, since you are mutating the object with push.

Reply:
done - added an action "addProjectedBudget"